File: typescript/last_word_length.ts
class LastWordLength{
    constructor(private readonly s:string){
        this.s = s
    }

    lastLength(){
        let length: number = 0;
        for(let i=this.s.length -1; i>0; --i){
            if(this.s[i] !== ' ') length += 1
            if(length > 0 && this.s[i] === ' ') return length
        }
        return length
    }
}

const lengthObj = new LastWordLength('   Hello    World  hello  ')
console.log(lengthObj.lastLength())

File: python/Easy/last_word_length.py
class LastWordLength:
    def __init__(self, s:str):
        self.s = s

    def lastLength(self):
        # split_str = [str for str in self.s.split(' ') if str != '']
        # return len(split_str[-1])
        length = 0
        for i in range(len(self.s) - 1, -1, -1):
            if self.s[i] != ' ':
                length += 1
            if length > 0 and self.s[i] == ' ':
                return length

lengthObj = LastWordLength('Hello World')
print(lengthObj.lastLength())